File: python/guidos-gorgeous-lasagna/lasagna.py
EXPECTED_BAKE_TIME = 40

LAYER_PREPARATION_TIME = 2

def bake_time_remaining(elapsed_bake_time):
    """Calculate the bake time remaining.

    :param elapsed_bake_time: int - baking time already elapsed.
    :return: int - remaining bake time (in minutes) derived from 'EXPECTED_BAKE_TIME'.

    Function that takes the actual minutes the lasagna has been in the oven as
    an argument and returns how many minutes the lasagna still needs to bake
    based on the `EXPECTED_BAKE_TIME`.
    """

    return EXPECTED_BAKE_TIME - elapsed_bake_time

def preparation_time_in_minutes(layers):
    """Calculate the required time to bake a lasagna with a given number of layers.

    :param layers: int - the number of layers in lasagna
    :return: int - time required to prepare lasagna
    """

    return layers * LAYER_PREPARATION_TIME

def elapsed_time_in_minutes(layers, elapsed_baking_time):
    """Calculate the total elapsed time of a lasagna that is baking.

    :param layers: int - the number of layers in lasagna
    :param elapsed_baking_time: int - the number of minutes the lasagna has spent baking
    :return: int - total elapsed time in preparation and baking of lasagna
    """

    return preparation_time_in_minutes(layers) + elapsed_baking_time

File: php/lasagna/Lasagna.php
<?php

class Lasagna
{
    // A lasagna takes 40 minutes to prep
    public function expectedCookTime()
    {
        return 40;
    }

    // @param int $elapsed_minutes The amount of time the lasagna has spent in the oven
    // @return int The remaining time left to finish cooking
    public function remainingCookTime(int $elapsed_minutes): int
    {
        return $this->expectedCookTime() - $elapsed_minutes;
    }

    // A lasagna's layers take 2 minutes to prep, each
    //
    // @param int $layers_to_prep The number of layers in the lasagna
    // @return int The number of minutes of prep
    public function totalPreparationTime(int $layers_to_prep): int
    {
        return $layers_to_prep * 2;
    }

    // The total time that has been spent on a lasangna that is in the oven
    //
    // @param int $layers_to_prep The number of layers in the lasagna that have been prepped\
    // @param int $elapsed_minutes The amount of time the lasagna has spent in the oven
    //
    // @return int The total amount of time spent in the lasagna (in minutes)
    public function totalElapsedTime(int $layers_to_prep, int $elapsed_minutes): int
    {
        return $this->totalPreparationTime($layers_to_prep) + $elapsed_minutes;
    }

    // @return 'Ding!'
    public function alarm(): string
    {
        return 'Ding!';
    }
}
